ts/src/kucoin.ts

```diff
@@ -1071,41 +1044,39 @@ export default class kucoin extends Exchange {
             const fee = this.safeNumber (entry, 'withdrawalMinFee');
             const active = (isWithdrawEnabled && isDepositEnabled);
             const networks = {};
-            const chains = this.safeValue (currencyChains, id, []);
+            const chains = this.safeValue (entry, 'chains', []);
+            const precision = this.parseNumber (this.parsePrecision (this.safeString (entry, 'precision')));
             for (let j = 0; j < chains.length; j++) {
                 const chain = chains[j];
                 const chainId = this.safeString (chain, 'chain');
-                const isChainEnabled = this.safeString (chain, 'isChainEnabled'); // better than 'status'
-                if (isChainEnabled === 'true') {
-                    const networkCode = this.networkIdToCode (chainId);
-                    const chainWithdrawEnabled = this.safeValue (chain, 'isWithdrawEnabled', false);
-                    const chainDepositEnabled = this.safeValue (chain, 'isDepositEnabled', false);
-                    networks[networkCode] = {
-                        'info': chain,
-                        'id': chainId,
-                        'name': this.safeString2 (chain, 'chainFullName', 'chainName'),
-                        'code': networkCode,
-                        'active': chainWithdrawEnabled && chainDepositEnabled,
-                        'fee': this.safeNumber (chain, 'withdrawMinFee'),
-                        'precision': this.parseNumber (this.parsePrecision (this.safeString (chain, 'walletPrecision'))),
-                        'limits': {
-                            'withdraw': {
-                                'min': this.safeNumber (chain, 'withdrawMinSize'),
-                                'max': undefined,
-                            },
-                            'deposit': {
-                                'min': this.safeNumber (chain, 'depositMinSize'),
-                                'max': undefined,
-                            },
+                const networkCode = this.networkIdToCode (chainId);
+                const chainWithdrawEnabled = this.safeValue (chain, 'isWithdrawEnabled', false);
+                const chainDepositEnabled = this.safeValue (chain, 'isDepositEnabled', false);
+                networks[networkCode] = {
+                    'info': chain,
+                    'id': chainId,
+                    'name': this.safeString (chain, 'chainName'),
+                    'code': networkCode,
+                    'active': chainWithdrawEnabled && chainDepositEnabled,
+                    'fee': this.safeNumber (chain, 'withdrawalMinFee'),
+                    'precision': precision,
```

Comment:
```suggestion
                    'precision': undefined,
```
 this should be, because that precision is not "network specific"